GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -752,34 +752,51 @@ private void UpdatePluginMenu(bool validWorkingDir)
 
         private void RegisterPlugins()
         {
+            const string PluginManagerName = "Plugin Manager";
             var existingPluginMenus = pluginsToolStripMenuItem.DropDownItems.OfType<ToolStripMenuItem>().ToLookup(c => c.Tag);
 
             lock (PluginRegistry.Plugins)
             {
                 var pluginEntries = PluginRegistry.Plugins
                     .OrderBy(entry => entry.Name, StringComparer.CurrentCultureIgnoreCase);
 
+                // the menu already contains 2 items:
+                //    [0] Separator
+                //    [1] Plugin Settings
+                // insert plugins above the separator (the list is in reverse order)
```

Comment:
@RussKie There is this comment that I don't understand.

False or not clear?
Which list is in reverse order? Because the menu items and the plugins **are not** in reversed order...

I proposed a change (https://github.com/gitextensions/gitextensions/pull/8238#discussion_r441880013 ) that reverse the plugins to simplify the code. But I don't know if it handles well edge cases (like duplicate of `PluginManager` in the Application folder and also in the user folder. I don't even know how it is handled now....).

Reply:
To be clearer, is it possible to have 2 plugins with the same name?

Reply:
It is certainly not impossible, I don't believe we guard anywhere against this.

> Which list is in reverse order? Because the menu items and the plugins **are not** in reversed order...

`pluginsToolStripMenuItem.DropDownItems` is a stack, so if we use `Add` we would add to the top. But here we use `Insert` to add from the other side of it. That's was "reverse" when I initially wrote that comment. But since then the code has changed few times, and this comment is probably no longer relevant, and should be removed.